piccolo/table.py

```diff
@@ -851,6 +851,13 @@ def __repr__(self) -> str:
         )
         return f"<{self.__class__.__name__}: {pk}>"
 
+    def __eq__(self, other: Table):
+        return (
+            isinstance(other, self.__class__)
+            and self.__class__._meta.primary_key
+            == other.__class__._meta.primary_key
```

Comment:
So here we're comparing two foreign key columns. But we need to be looking at their values.

Reply:
i.e. `getattr(self, self.__class__._meta.primary_key._meta.name)`.

Reply:
I also think that this solution is OK, because, as you said, we need values
```python
def __eq__(self, other) -> bool:
    return isinstance(other, self.__class__) and getattr(
        self, self._meta.primary_key._meta.name, None
    ) == getattr(other, other._meta.primary_key._meta.name, None)
```
because it satisfies both the correct results and `mypy`. I wrote some tests and they passed.